src/cockpit/channels/filesystem.py

```diff
@@ -50,6 +52,12 @@
 logger = logging.getLogger(__name__)
 
 
+@functools.lru_cache()
+def my_umask() -> int:
+    match = re.search(r'^Umask:\s*0([0-7]*)$', Path('/proc/self/status').read_text(), re.M)
```

Comment:
Ugh, this is horrible.. I know that [os.umask](https://docs.python.org/3/library/os.html#os.umask) is weird too (due to `umask(2)`). Calling [getumask()](https://www.man7.org/linux/man-pages/man3/getumask.3.html) via ctypes is equally involved, though. And NamedTemporaryFile() doesn't have a mode where it would just open the file normaly. At times, Linux APIs are really shitty..

Reply:
The ctypes call isn't bad.  I knew there was nothing at the kernel level for this, but a threadsafe-via-lock equivalent in the C library isn't bad!  Is that on RHEL8?

Reply:
```
⬢[lis@toolbox ~]$ podman run -it ubi8
[root@c4818a63b310 /]# dnf install python3
...
[root@c4818a63b310 /]# python3
Python 3.6.8 (default, Jan  5 2024, 08:58:17) 
[GCC 8.5.0 20210514 (Red Hat 8.5.0-20)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ctypes
>>> c = ctypes.CDLL(None)
>>> c.getumask()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.6/ctypes/__init__.py", line 356, in __getattr__
    func = self.__getitem__(name)
  File "/usr/lib64/python3.6/ctypes/__init__.py", line 361, in __getitem__
    func = self._FuncPtr((name_or_ordinal, self))
AttributeError: python3: undefined symbol: getumask
```

bzzt

Reply:
It was more of a lament than a request to change, but thanks for trying :grin:

Reply:
It doesn't work on Fedora 39 either, btw